autoload/unite/sources/decls.vim

```diff
@@ -0,0 +1,68 @@
+let s:save_cpo = &cpoptions
+set cpoptions&vim
+
+let s:source = {
+      \ 'name': 'decls',
+      \ 'description': 'GoDecls implementation for unite',
+      \ 'syntax': 'uniteSource__Decls',
+      \ 'action_table': {},
+      \ 'hooks': {},
+      \ }
+
+function! unite#sources#decls#define()
+  return s:source
+endfunction
+
+function! s:source.gather_candidates(args, context) abort
+  let l:bin_path = go#path#CheckBinPath('motion')
+  if empty(l:bin_path)
+    return []
+  endif
+
+  let l:path = expand(get(a:args, 0, '%:p:h'))
+  if isdirectory(l:path)
+    let l:mode = 'dir'
+  elseif filereadable(l:path)
+    let l:mode = 'file'
+  else
+    return []
+  endif
+
+  let l:include = get(g:, 'go_decls_includes', 'func,type')
+  let l:command = printf('%s -format vim -mode decls -include %s -%s %s', l:bin_path, l:include, l:mode, l:path)
```

Comment:
You should `shellescape()` most or all the variables you're printf-ing in the command string; otherwise stuff will break if the path contains spaces or special shell characters (this is uncommon in Go, but not illegal either, and especially an issue on systems like Windows with stuff `Application Data` in the path).

For example, if I use `:Unite decls` on `~/go/src/a/space path/space name.go` I get `[unite.vim] [decls] command returned invalid response.`.

Reply:
Thx for reviewing 👍 I fixed it.